### PR TITLE
feat(inputs.win_perf_counters): Add remote computers support

### PR DESCRIPTION
**Pull request by sspaink**
Replacing: https://github.com/influxdata/telegraf/pull/6590
Resolves https://github.com/influxdata/telegraf/issues/1899

At the moment we are waiting for customer/community participation to help ensure this change meets their needs (configuration makes sense, etc.). Therefore we will need to be able to re-base it and get new build artifacts on request, moving this pull request to be under a local branch seems like the best approach.

# original description:

Added support for reading performance counters from remote computers. User, under who is Telegraf running, must be already authenticated to the remote computer(s).

The support is added via additional optional global parameter `Sources`. This parameter can be also set for performance object to configure per host-specific performance counter:
``` toml
[[inputs.win_perf_counters]]
  PreVistaSupport=false
  UseWildcardsExpansion=false
  UsePerfCounterTime=true
  CountersRefreshInterval="0"
  PrintValid=true
  # Names or ip addresses of remote computers to gather counters from, including local computer.
  # Telegraf's user must be already authenticated to the remote computers.
  # It can be overridden at the object level
  Sources=["localhost", "192.168.10.12"]

[[inputs.win_perf_counters.object]]
   # Overrides the Sources global parameter for current performance object.
    Sources = ["192.168.1.68"]
    # Processor usage, alternative to native, reports on a per core.
    ObjectName = "Processor"
    Instances = ["*"]
    Counters = [
      "% Processor Time",
    ]
    Measurement = "win_cpu"
```
It allows to define the same configuration for multiple computers and also different configurations per computer.

Computer name is then added as the `source` tag. Telegraf host compuper has the `source` set to the computer name.

Readme has been also enhanced with plugins's schema description.

### Required for all PRs:
- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [X] Has appropriate unit tests.

### CODE REVIEW

**Review by srebhan — COMMENTED** [review]
Thanks for implementing this long-standing request @sspaink ! I have some suggestions in the code and one request to prefer `require` over `assert`.

**Comment by reimda** [comment]
@cruscio I think you and Allscripts have been interested in this feature for some time now (https://github.com/influxdata/telegraf/issues/8238). Would you be able to test the build artifacts of this PR posted by telegraf-tiger? Once this PR passes code review we would like to merge it but only if it meets users' needs, including yours. Thanks!

**Comment by deweter** [comment]
Please, reopen this PR. We are very interested in this.
Thank you!

**Comment by sspaink** [comment]
@deweter I can re-open this and get it ready for review, would you be able to help test this feature and provide feedback on the config to help make sure it is ready?

**Comment by deweter** [comment]
Yes, we would. At this moment, we are using telegraf in env with 500+ servers. Now it has to be deployed locally just because perfmon counters cannot be gathered remotely.

**Comment by sspaink** [comment]
@deweter Whenever you have time, could you try the artifacts posted by the tiger bot and see if these new feature works for you? And let me know if the documentation is enough: https://github.com/influxdata/telegraf/blob/vh-wpc-remote-computers-support/plugins/inputs/win_perf_counters/README.md

Thank you!

**Comment by sspaink** [comment]
@deweter Have you had time to test out this feature? The artifacts will expire in the next 8 days.

**Comment by deweter** [comment]
I am sorry for the delay. Please, bare with me, I am going to have a look at it within next two weeks.

**Comment by sspaink** [comment]
No worries, I've re-based so now the artifacts posted will last another month.

**Comment by hpainter2002** [comment]
Hello, can this PR be reopened? I am interested in this feature. I can provide support needed to get this merged. Thank you!

**Comment by MyaLongmire** [comment]
@sspaink are you still willing to work on this pr?

**Comment by sspaink** [comment]
@hpainter2002 thank you your help would be appreciated, can you try out one of the artifacts the Tiger bot posted to see if this feature works for you? If you could also provide feedback on the documentation in this PR that would be great.

**Comment by hpainter2002** [comment]
@sspaink Yes, I can try out the artifact. I will let you know if the feature is sufficient.

File: plugins/inputs/win_perf_counters/README.md [hunk 1]
```diff
@@ -124,6 +138,23 @@ errors](pdh.go).
...
+
+(Optional)
+
+Host names or ip addresses of computers to gather all performance counters from.
+The user running Telegraf must be authenticated to the remote computer(s).
+E.g. via Windows sharing `net use \\SQL-SERVER-01`.
+Use either localhost (`"localhost"`) or real local computer name to gather counters
+also from localhost among other computers. Skip, if gather only from localhost.
+
+If a performance object is present only on specific hosts set `Source` param on
```
Comment (hpainter2002) [inline plugins/inputs/win_perf_counters/README.md:151]:
```suggestion
If a performance counter is present only on specific hosts set `Sources` param on
```

File: plugins/inputs/win_perf_counters/README.md [hunk 2]
```diff
@@ -124,6 +138,23 @@ errors](pdh.go).
...
+the object level configuration to override global Sources.
```
Comment (hpainter2002) [inline plugins/inputs/win_perf_counters/README.md:152]:
```suggestion
the specific counter level configuration to override global (plugin wide) Sources.
```

File: plugins/inputs/win_perf_counters/README.md [hunk 3]
```diff
@@ -181,6 +212,13 @@ This must be specified for every counter you want the results of, or use
 `["*"]` for all the counters of the object, if the `UseWildcardsExpansion` param
 is set to `true`.
 
+#### Sources (Object)
+
+(Optional)
+
+Overrides the [Sources](#sources) global parameter for current performance object.
+See [Sources](#sources) description for more details.
+
```
Comment (hpainter2002) [inline plugins/inputs/win_perf_counters/README.md:221]:
```suggestion
Example: 
 ` [[inputs.win_perf_counters.object]]
    ObjectName = "Memory"
    Counters = ["Available Bytes"]
    Instances = ["------"]
    Measurement = "win_mem"
    Sources = ["localhost"]`
```

@sspaink Having an example on how to override is helpful in my opinion, let me know your thoughts.

**Comment by hpainter2002** [comment]
@sspaink I have tested [windows_amd64.zip](https://output.circle-artifacts.com/output/job/6529e4b3-c659-4dc8-b3e8-552cafc5a3f5/artifacts/0/build/dist/telegraf-1.25.0~22bf0c9c_windows_amd64.zip). Seems to work for us. I have also added a few suggestions on the documentation (Let me know if you would prefer me to start a review). Feel free to update as you see fit. Thank you!

File: plugins/inputs/win_perf_counters/README.md (hunk 3 above)
Comment (srebhan) [reply plugins/inputs/win_perf_counters/README.md:221]: We can add this later I think.

**Comment by srebhan** [comment]
Superseeded by #12556.

**Pull request closed without merging**